modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/RestMultiSearchTemplateAction.java

```diff
@@ -79,6 +79,14 @@ public String getName() {
     @Override
     public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
         MultiSearchTemplateRequest multiRequest = parseRequest(request, allowExplicitIndex);
+
+        // Emit a single deprecation message if any search template contains types.
+        for (SearchTemplateRequest searchTemplateRequest : multiRequest.requests()) {
+            if (searchTemplateRequest.getRequest().types().length > 0) {
+                deprecationLogger.deprecatedAndMaybeLog("msearch_with_types", TYPES_DEPRECATION_MESSAGE);
+                break;
+            }
+        }
```

Comment:
Can this be unit tested, since a similar looking test is removed from MultiSearchTemplateRequestTests? Maybe it already is tested elsewhere, but I don't see it in this PR so just asking to check to make sure.

Reply:
Yes, this deprecation warning is tested in `RestMultiSearchTemplateActionTests`. The `assertWarnings` call in `MultiSearchTemplateRequestTests` wasn't there to explicitly test the deprecation, but rather to make sure the test didn't fail on unexpected warnings.

Reply:
Thanks, and sorry for bothering.